backend/services/userProfile.service.test.js

```diff
@@ -0,0 +1,89 @@
+const userProfileService = require('./userProfile.service');
+const modificationLogService = require('./modificationLog.service');
+const DatabaseError = require('../errors/database.error');
+
+const { setupDB } = require("../setup-test");
+
+setupDB("userProfile-service");
+
+describe("UserProfileService can save/update/get user profiles", () => {
+  test("Save a user profile record and then retrieve it by the service", async (done) => {
+
+
+    const newRecordData = {signupEmail: "FOO@BAR.COM" };
+
+    const result = await userProfileService.createUser(newRecordData, "creator@bar.com");
+
+    expect(result.isNew).toBe(false);
+
+    // Verify email address was lowercased
+    expect(result.signupEmail).toBe(newRecordData.signupEmail.toLowerCase());
+
+    // verify log record was created
+    const logResult = await modificationLogService.getLogs(result._id, "UserProfile");
+
+    expect(logResult.length).toBe(1);
+
+    const findUserResult = await userProfileService.getUser(result._id);
+    expect(findUserResult.signupEmail).toBe(result.signupEmail);
+
+    const findUserByEmailResult = await userProfileService.getUserByEmail(result.signupEmail);
+    expect(findUserByEmailResult.signupEmail).toBe(result.signupEmail);
+
+    done();
+  });
+
+  test("Try and create a user profile with no email. Should throw an exeception", async (done) => {
+
+    const newRecordNoEmailDefined = {};
+    const newRecordEmailNull= {signupEmail: null};
+    const newRecordNoEmail = {signupEmail: "" };
+
+    await expect(userProfileService.createUser(newRecordNoEmailDefined))
+      .rejects.toThrow(DatabaseError);
+    await expect(userProfileService.createUser(newRecordEmailNull)).rejects.toThrow(DatabaseError);
+    await expect(userProfileService.createUser(newRecordNoEmail)).rejects.toThrow(DatabaseError);
+
+    done();
+  });
+
+  test("Try and update a user profile", async (done) => {
```

Comment:
Please rename the test from `Try and update a user profile` to `Should update a user profile with allowed values. Should not allow update email.`